### PR TITLE
Check all err, fix ineffassign and structure files via goimports

### PR DESCRIPTION
**Pull request by njuettner**
* Use golangci-lint

### CODE REVIEW

File: pkg/updatestrategy/drain_test.go [hunk 1]
```diff
@@ -24,7 +24,10 @@ const (
 
 func removePod(client kubernetes.Interface, pod v1.Pod) error {
 	var zero int64
-	client.CoreV1().Pods(pod.GetNamespace()).Delete(pod.GetName(), &metav1.DeleteOptions{GracePeriodSeconds: &zero})
+	err := client.CoreV1().Pods(pod.GetNamespace()).Delete(pod.GetName(), &metav1.DeleteOptions{GracePeriodSeconds: &zero})
```
Comment (linki) [inline pkg/updatestrategy/drain_test.go]: If the pod doesn't exist then this returns an `err` here. Looks like the original code would like to just ignore it.

File: provisioner/clusterpy.go [hunk 2]
```diff
@@ -294,7 +294,7 @@ func (p *clusterpyProvisioner) Provision(ctx context.Context, logger *log.Entry,
 	// accounts.
 	bucketName := fmt.Sprintf(clmCFBucketPattern, strings.TrimPrefix(cluster.InfrastructureAccount, "aws:"), cluster.Region)
 
-	err = createOrUpdateClusterStack(ctx, awsAdapter, cfgBasePath, cluster, values, bucketName)
+	err = createOrUpdateClusterStack(awsAdapter, ctx, templateCtx, path.Join(configPath, clusterStackFileName), bucketName)
```
Comment (szuecs) [inline provisioner/clusterpy.go]: ctx is normally always the first parameter, a change looks weird to me

**Comment by linki** [comment]
:+1:

**Comment by mikkeloscar** [comment]
I think we need to checkin generated cluster registry client for golangci-lint to work:

```
$ GOLANGCI_COM_RUN=1 golangci-lint run --out-format=json --issues-exit-code=0 --deadline=5m --new=false --new-from-rev= --new-from-patch=../changes.patch
    Running error: context loading failed: package github.com/zalando-incubator/cluster-lifecycle-manager/cmd/clm
    	imports github.com/zalando-incubator/cluster-lifecycle-manager/controller
    	imports github.com/zalando-incubator/cluster-lifecycle-manager/registry
    	imports github.com/zalando-incubator/cluster-lifecycle-manager/pkg/cluster-registry/client: cannot find package "github.com/zalando-incubator/cluster-lifecycle-manager/pkg/cluster-registry/client" in any of:
    	/go/src/github.com/zalando-incubator/cluster-lifecycle-manager/vendor/github.com/zalando-incubator/cluster-lifecycle-manager/pkg/cluster-registry/client (vendor tree)
    	/usr/local/go/src/github.com/zalando-incubator/cluster-lifecycle-manager/pkg/cluster-registry/client (from $GOROOT)
    	/go/src/github.com/zalando-incubator/cluster-lifecycle-manager/pkg/cluster-registry/client (from $GOPATH): failed to analyze
```

**Comment by arjunrn** [comment]
> I think we need to checkin generated cluster registry client for golangci-lint to work:

Let's do that in a followup PR because this PR has been open long enough. Also I'm not sure about the security implications considering the `cluster-registry` is private.

**Comment by arjunrn** [comment]
:+1:

**Comment by mikkeloscar** [comment]
> Also I'm not sure about the security implications considering the cluster-registry is private.

The code is generated at build-time based on the swagger spec which we have included here. So it's just a matter of checking in what we already are building.

**Pull request merged**